src/OpenTelemetry/Trace/Export/IReadableSpan.cs

```diff
@@ -0,0 +1,81 @@
+﻿// <copyright file="IReadableSpan.cs" company="OpenTelemetry Authors">
+// Copyright 2018, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using OpenTelemetry.Resources;
+
+namespace OpenTelemetry.Trace.Export
+{
+    public interface IReadableSpan
```

Comment:
I think this would be clearer if named `IReadOnlySpan`.

Reply:
It would conflict with [ReadOnlySpan](https://docs.microsoft.com/en-us/dotnet/api/system.readonlyspan-1?view=netcore-3.1) 

Also ReadableSpan term was used in spec and used in [java](https://github.com/open-telemetry/opentelemetry-java/blob/023f3fef2502f270d17c9e3c6c128d92d378ec90/sdk/src/main/java/io/opentelemetry/sdk/trace/ReadableSpan.java) and might be in other languages

Reply:
`IReadOnlySpan` (interface) won't conflict with `ReadOnlySpan` (System.Memory). I do see there is going to be confusion between OpenTelemetry and System.Memory but I don't think we can avoid all of it.

My preference is still `IReadOnlySpan`. I don't think *Readable* means anything where *ReadOnly* is a more generally accepted concept.

@SergeyKanzhelev @cijothomas @bruno-garcia what do you think?

Reply:
Agree `ReadableSpan` isn't a great name and it's true that `IReadOnlySpan` wouldn't conflict with `ReadOnlySpan` (`I` in there) even though I got confused for a sec when I saw the name mentioned here. 

That said, `IReadOnlySpan` would be more idiomatic for .NET devs albeit confusing a bit with _the other ReadOnlySpan_. There's the cost of being foreign to people familiar with OTel (assuming `ReadableSpan` gets picked up by most other SDKs).

That said, I have no strong opinions here but just for the sake of giving one: 
In this case, it could be worth sticking with the name that one might encounter while reading up on OTel more generally, so `IReadableSpan`.

Reply:
done a bit of research and noticed that several OTel SDKs use `SpanData` except for those which use `Span` impl directly and do not have any Readable/SpanData notion. `ReadableSpan` seems to be used by Java only.

I believe `ISpanData` may be a good alternative as it is something common across SDKs and does not conflict with very common .NET `ReadOnlySpan`. Event `ISpan` is a problem (but seems like renaming it has a lot of drawbacks) but introducing yet another conflict seems unreasonable.

@MikeGoldsmith @SergeyKanzhelev @bruno-garcia ?

Reply:
SpanData is a good name

Reply:
👍 for `SpanData`